File: Programmers/L2/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int count = 0, max = 0;
    for(int i = 0; i < progresses.size(); i++) {
        int asu = (100-progresses[i])/speeds[i];
        if(((100-progresses[i])%speeds[i]))
            ++asu;
        
        if(i == 0) {
            max = asu;
            count++;
        } else if(i == progresses.size()-1) {
            if(max < asu) {
                answer.emplace_back(count);
                count = 1;
            } else{
                count++;
            }
            answer.emplace_back(count);
        } else {
            if(max < asu) {
                answer.emplace_back(count);
                max = asu;
                count = 1;
            } else {
                count++;
            }
        }
        
    }
    
    return answer;
}

File: Programmers/JAVASCRIPT/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let cnt = 1;
    let before = 0;
    let temp = 0;
    before = parseInt((100 - progresses[0]) / speeds[0]);
    for(let i = 1; i < progresses.length; i++) {
        temp = parseInt((100 - progresses[i]) / speeds[i]);
        if(before >= temp) 
            cnt++;
        else {
            answer.push(cnt);
            before = temp;
            cnt = 1;
        }
    }
    answer.push(cnt);
    return answer;
}